lib/logstash/outputs/elasticsearch.rb

```diff
@@ -413,11 +415,17 @@ def submit(actions)
       @submit_mutex.unlock
     end
     if bulk_response["errors"]
-      failed_actions = actions.select.with_index {|_,i| [429, 503].include?(bulk_response['statuses'][i]) }
-      unless failed_actions.empty?
-        @logger.debug "#{failed_actions.size}/#{actions.size} events were unsuccessful in sending"
-        retry_push(failed_actions)
+      actions_with_responses = actions.zip(bulk_response['statuses'])
+      actions_to_retry = []
+      actions_with_responses.each do |action, resp_code|
```

Comment:
If I'm reading this correctly, this is to handle "partial retry" differently than "full retry" ? Where RETRYABLE_CODES indicates a partial retry and not being in SUCCESS_CODES means a full retry?